spec.md

```diff
@@ -419,25 +421,7 @@ the set of known extensions - not just for possible name conflicts but for
 extensions that might be of interest.
 
 Each specification that defines how to serialize a CloudEvent will define how
-extension attributes will appear.
-
-Extension attribtue MUST be serialized using the same general pattern as all
```

Comment:
why did you drop this section? It's kind of the most important pieces from #508

Reply:
It's from #505 not #508.

Extensions should follow the same patterns in the sense of the abstract event model - i.e. use the same type system.

However event encodings can't  be required to follow "the same patterns" in how they encode extensions, since this spec doesn't have any notion of any event encoding,  so it can't say what "patterns" might be best.

The AMQP, HTTP et. al. specs do say what those encodings look like, and of course it makes sense to follow similar patterns when designing a new binding or format, but it doesn't make sense to mandate that in *abstract* spec. 

E.g. the idea of "alternate locations" is used in HTTP and AMQP but not JSON. The requirement that extensions MUST be encoded using "similar patterns" to standard attributes isn't even followed by existing HTTP and AMQP binary encodings - they use a special "alternate location" pattern for datacontenttype, which can't possibly be followed for extension attributes. The original JSON format put extensions in a separate map - som future binding might need to do the same.

In short, a binding MUST be able to faithfully encode and decode an abstract event to/from its internal representation - but how it does so is for the binding spec.

Reply:
I guess it depends on whether there is value in the abstract spec providing the overall guidance that all binding specs need to follow.  If the group decides that it's ok to leave it up to each binding/transport spec, then we'd need to define "secondary" in each spec because right now those other specs point to the main spec for that definition.

What do others think?  Leave it up to the binding specs or have a central statement about the consistency?

Reply:
It's fine to have examples or suggestions, but not in normative language. The spec needs to be clear on what is absolutely required of a binding, and not confuse that with suggestions about what might be a good idea for some (or even most) bindings.

Reply:
It's not an example though, it says that extensions are serialized the same way as normal CE attributes. Yes each transport might define its own way of doing that serialization but I think its helpful to say (at the spec level) that all attributes (core and extensions) are equal and treated as such.  Then each transport can define what those generic rules are for that specific transport.

Reply:
CE attributes are *not* all serialized the same way, not even core attributes in our existing bindings. datacontenttype is treated differently in AMQP and HTTP because they happen to have a spot that clearly should carry the content-type. We don't know what oddities future protocol bindings might have, so I don't see any value in MUSTy language on the subject. As long as a binding faithfully represents the abstract CE event, there's no need for the core spec to say how it should be done.

Reply:
Concrete example: the AMQP CE spec chose to map datacontentype to AMQP message property content-type. It chose *not* to map CE "id" and "time" attributes to AMQP message properties message-id and creation-time. I agree with those choices, but it would be a mistake to force similar choices on future bindings from the core spec; they should be informed by the context of the protocol we are binding to.

Reply:
Put back a modified version of this now I get your concern.